selfdrive/ui/sunnypilot/qt/network/networking.cc

```diff
@@ -43,3 +50,35 @@ NetworkingSP::NetworkingSP(QWidget *parent) : Networking(parent) {
   )");
   setStyleSheet(newStyleSheet);
 }
+
+AdvancedNetworkingSP::AdvancedNetworkingSP(QWidget *parent, WifiManager *wifi) : AdvancedNetworking(parent, wifi) {
+  // Persist tethering toggle
+  auto tetheringPersistToggle = new ToggleControl(tr("Persist tethering across reboot"), "", "", params.getBool("PersistTetheringAcrossReboot"));
+  connect(tetheringPersistToggle, &ToggleControl::toggleFlipped, [=](bool state) {
+    params.putBool("PersistTetheringAcrossReboot", state);
+  });
+  // Insert it after tetheringToggle
+  auto buttons = findChildren<QPushButton*>();
+  auto tetheringToggleTitle = std::find_if(buttons.begin(), buttons.end(), [](QPushButton* btn) {
+    return btn->text() == tr("Enable Tethering");
+  });
+  auto tetheringToggle = tetheringToggleTitle != buttons.end() ? (*tetheringToggleTitle)->parentWidget() : nullptr; // Toggle widget is the parent of the button label
+  auto list = findChild<ListWidgetSP*>();
+  if (tetheringToggle && list) {
+    list->AddWidgetAt(list->indexOf(tetheringToggle) + 1, tetheringPersistToggle);
+  }
+
+  // Initialize tethering on startup
+  if (params.getBool("TetheringEnabled") && params.getBool("PersistTetheringAcrossReboot")) {
+    QTimer::singleShot(2000, this, [=] {
+      qDebug() << "Re-enabling tethering on startup";
+      wifi->setTetheringEnabled(true);
+      tetheringToggle->setEnabled(true);
```

Comment:
hm, this doesn't seem to work and I don't know Qt well enough to know why - afaik `tetheringToggle` is correctly targeting the `ToggleControlSP` (since it's correctly targeting it and adding our new toggle underneath it in the layout), is there some explicit re-render I need to call to get this to refresh visually?